javascript/javascript1/week2/main.js

```diff
@@ -99,3 +99,48 @@ function todaysTemperature(temperature) {
   }
 }
 todaysTemperature(8);
+
+// Student manager
+const class07Students = [];
+function addStudentToClass(studentName) {
+  let i = 0;
+  for (; i < class07Students.length; i++) {
+    if (studentName === class07Students[i]) {
+      console.log("Student " + studentName + " is already in the class.");
+      break;
+    } else {
+      continue;
+    }
+  }
+  if (i === class07Students.length) {
+    if (class07Students.length > 5) {
+      if (studentName === "Queen") {
+        class07Students.push(studentName);
+        console.log(class07Students);
+      } else {
+        console.log("Cannot add more students to class 07");
+      }
+    } else {
+      class07Students.push(studentName);
+      console.log(class07Students);
+    }
+  }
+}
+
+addStudentToClass("Rismita");
+addStudentToClass("Rismita");
+addStudentToClass("Abhirup");
+addStudentToClass("Aaditri");
+addStudentToClass("Amrita");
+addStudentToClass("Neo");
+addStudentToClass("Neo");
+addStudentToClass("Leo");
+addStudentToClass("Sudip");
+addStudentToClass("Queen");
+addStudentToClass("Uma");
+
+function getNumberOfStudents() {
+  let studentStrength = class07Students.length;
+  console.log(studentStrength);
+}
+getNumberOfStudents();
```

Comment:
Once again, great work. 👌
It can be tricky to keep track of nested if else statements and you have done really well. ✨

**There are a few things to consider for future implementations**
Normally you will use statement 1 in the for loop to initialize the variable used in the loop, as the example below;
    -  for (let i = 0; i < class07Students.length; i++)

I can see, that you are using the variable i, in an if statement outside of the for loop scope, which is probably why you initialized it prior to the for loop. It shows great creativity and understanding of scopes. 👌
Try to experiment with making the code even cleaner and simpler by shifting out the nested if else statements with an if, else if, else structure.